File: test/fixtures/node-axios/j_post_form_f.js
import axios from 'axios';
import FormData from 'form-data';

const form = new FormData();
form.append('d1', 'data1');
form.append('d2', 'data');

const response = await axios.post(
  'http://localhost:28139/post',
  form,
  {
    headers: {
      ...form.getHeaders()
    }
  }
);

File: test/fixtures/kotlin-ktor/j_post_form_f.kt
import io.ktor.client.HttpClient
import io.ktor.client.request.post
import io.ktor.client.request.setBody
import io.ktor.client.statement.bodyAsText
import io.ktor.http.isSuccess
import kotlinx.coroutines.runBlocking

val client = HttpClient()

runBlocking {
    val response = client.post("http://localhost:28139/post") {
        setBody(
    MultiPartFormDataContent(
        formData {
                append("d1", "data1")
                append("d2", "data")
            }
    )
)
    }
    if (!response.status.isSuccess()) throw Exception("Request failed with status code ${response.status}")

    println(response.bodyAsText())
}
